src/boot/bootctl.c

```diff
@@ -1860,13 +1860,22 @@ static int verb_systemd_efi_options(int argc, char *argv[], void *userdata) {
         int r;
 
         if (argc == 1) {
-                _cleanup_free_ char *line = NULL;
+                _cleanup_free_ char *line = NULL, *new = NULL;
 
                 r = systemd_efi_options_variable(&line);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to query SystemdOptions EFI variable: %m");
-
-                puts(line);
+                if (r == -ENODATA)
+                        log_debug("No SystemdOptions EFI variable present in cache.");
+                else if (r < 0)
+                        return log_error_errno(r, "Failed to read SystemdOptions EFI variable from cache: %m");
+                else
+                        puts(line);
+
+                r = systemd_efi_options_efivarfs_if_newer(&new);
+                if (r < 0 && r != -ENODATA)
+                        log_warning_errno(r, "Failed to check EFI variable in efivarfs, ignoring: %m");
+                if (new)
+                        log_notice("Note: SystemdOptions EFI variable has been modified since boot. New value: %s",
+                                   new);
```

Comment:
hmm, I think we should suppress any such notice if the string is 100% identical to the previous version, even if the mtime changed

Reply:
Done now.